File: CandyRainTolstonogova/CandyRainC++/CandyRain/GameObject.cpp
#include "GameObject.h"
#include <iostream>
GameObject::GameObject(int x, int y,  const std::string &sprite_path) :
    positionX(x), positionY(y), sprite(sprite_path) {
    std::cout << "Объект GameObject создан" << std::endl;
}

GameObject::~GameObject() {
    std::cout << "Объект GameObject уничтожен" << std::endl;
}
// Getter
int GameObject::getpositionX() const { return positionX; }
int GameObject::getpositionY() const { return positionY; }
std::string GameObject::getsprite() const { return sprite; }

// Setter
void GameObject::setpositionX(int count) { positionX = count; }
void GameObject::setpositionY(int count) { positionY = count; }
void GameObject::setsprite(std::string bg) { sprite = bg; }

void GameObject::displayInfo() const {
    std::cout << "GameObject: X=" << positionX << ", Y=" << positionY
        << ", Sprite=" << sprite << std::endl;
}

File: CandyRainTolstonogova/CandyRainJava/GameObject.java
public class GameObject {
    private int positionX;
    private int positionY;
    private String sprite;
    
    public GameObject(int x, int y, String spritePath) {
        this.positionX = x;
        this.positionY = y;
        this.sprite = spritePath;
        System.out.println("Объект GameObject создан");
    }
    
    // Getters
    public int getPositionX() { return positionX; }
    public int getPositionY() { return positionY; }
    public String getSprite() { return sprite; }
    
    // Setters
    public void setPositionX(int count) { this.positionX = count; }
    public void setPositionY(int count) { this.positionY = count; }
    public void setSprite(String bg) { this.sprite = bg; }
    
    public void displayInfo() {
        System.out.println("GameObject: X=" + positionX + ", Y=" + positionY + ", Sprite=" + sprite);
    }
}